lightning-background-processor/src/lib.rs

```diff
@@ -217,15 +220,21 @@ impl BackgroundProcessor {
 			let mut have_pruned = false;
 
 			loop {
+				// If we're awaiting graph sync to be completed, we do not want to modify the network graph
+				let is_currently_awaiting_graph_sync = is_awaiting_graph_sync_completion_clone.load(Ordering::Acquire);
+
 				peer_manager.process_events(); // Note that this may block on ChannelManager's locking
-				channel_manager.process_pending_events(&event_handler);
-				chain_monitor.process_pending_events(&event_handler);
+				if !is_currently_awaiting_graph_sync {
+					channel_manager.process_pending_events(&event_handler);
```

Comment:
Hmm, we definitely cant skip processing events while we're waiting on a download, I think - does this cause problems?

Reply:
`ProbabilisticScorer` will take a read lock on `NetworkGraph`, which would block, IIUC.

Reply:
Is that an issue? The update processing logic should take a few hundred ms, which is fine for the event processing time, I'd assumed most of the "awaiting" time here is just "waiting on download to complete", or was hoping so.

Reply:
The download + graph syncing time should indeed only take a few hundred ms max, but it needs a write lock to the network graph. So given the low time constraint, the delay seemed to be an acceptable trade-off.

Reply:
Actually, it looks like a write lock is taken per update, not overall, so maybe event processing would be fine?

Reply:
You mean by the graph sync process itself? That's true.

Reply:
My concern though remains that I don't want any of the handled events to trigger a change to the network graph, which definitely seems like a possibility. I guess an alternative would be propagating the awaiting boolean into the DecoratingEventHandler and filtering network events right there.

Reply:
That gets tricky because they need to be handled before we persist. If you really don't want it to change I think it would make more sense to take a write lock at the start of processing. That will block any event handling needing the lock, though.

Reply:
If network graph changes need to be handled before we can persist non-graph-related changes such as channel monitors, that once again brings us back to the point we're at right now, where we need to delay processing until graph sync completes, which is not an issue except in high-latency conditions.

Reply:
I don't really get why handling things which change the network graph before or after the initial sync changes anything? Like, we can always get HTLC failures that result in the network graph being changed, if that's going to throw off the next sync then ithat's a problem with the sync, not with the events. It'll either throw off the current sync or the next one, not much we can do about that.

Reply:
In that case, do we even really need this PR? Considering that the graph will settle into something reasonable quickly anyway, if a temporary partial sync disruption is acceptable, it might be a NO_OP.

Reply:
Yes - the concern is mass removal based on timeouts when we just have a stale timer for all our data. The concern over events is just a concern over removing things from the graph based on HTLC failures - that's a totally legit reason to remove something from the graph, even if the sync logic ends up with a duplicate removal later.

Reply:
In that case, I think the easiest path forward is modifying this PR to merely delay pruning the graph. Frankly, I think it's already a non-issue to begin with, considering the pruning starts only a minute in.

Reply:
Its not super likely to be an issue, sure, but its mobile - "shouldnt happen as long as the internet kinda works and can manage to fetch a file within a minute" is.....a dubious assumption :(

Reply:
Agreed. I think we might benefit from a battery that can detect faulty internet connectivity, because in high latency situations, the other events would take a while to show up, too, despite there perhaps already being a push notification. And user confusion/perceived loss/non-receipt of funds is the argument you mentioned above for not delaying event processing.